src/FuelClient.cc

```diff
@@ -719,17 +750,26 @@ Result FuelClient::DownloadModel(const ModelIdentifier &_id,
         std::string dependencyPath;
         ignition::common::URI dependencyURI(meta.dependencies(i).uri());
 
-        // If the model is not already cached, download it; this prevents
-        // any sort of cyclic dependencies from running infinitely
-        if (!this->CachedModel(dependencyURI, dependencyPath))
-          this->DownloadModel(dependencyURI, dependencyPath);
+        ModelIdentifier dependencyID;
+        this->ParseModelUrl(dependencyURI, dependencyID);
+        _dependencies.push_back(dependencyID);
       }
     }
   }
 
   return Result(ResultType::FETCH);
 }
 
+//////////////////////////////////////////////////
+Result FuelClient::DownloadModels(
```

Comment:
Is this function not finished? It doesn't appear to be doing anything and isn't being used